src/vaults/yields.utils.ts

```diff
@@ -463,3 +432,153 @@ function constructGraphVaultData(
+   * THE CURRENT YIELD SYSTEM.
+   *
+   * https://etherscan.io/tx/0x1e3e7c71012d36e936b768a37e9784125a00f205a22bd808f045968a506bb1ce#eventlog
+   *
+   * THIS TRANSACTION CONTAINS THE SINGLE BADGER TREE DISTRIBUTION WE ARE
+   * ALLOCATING TO GRAVI_AURA VAULT AS A MISSED - AND NOT CAPTURED SOURCE.
+   *
+   * THIS CODE SHOULD BE REMOVED BY 08/29.
+   */
+  if (vault.address === TOKENS.GRAVI_AURA) {
+    const targetBlock = 15344809;
+    const block = await sdk.provider.getBlock(targetBlock);
+    const targetedInsertion = recentHarvests[0].treeDistributions;
+    targetedInsertion.push({
+      timestamp: block.timestamp,
+      block: targetBlock,
+      token: TOKENS.BADGER,
+      amount: BigNumber.from('1928771715566995688546'),
+    });
+  }
+
+  const allHarvests = recentHarvests.flatMap((h) => h.harvests.map((h) => ({ ...h, type: HarvestType.Harvest })));
+  const allDistributions = recentHarvests.flatMap((h) =>
+    h.treeDistributions.map((d) => ({ ...d, type: HarvestType.TreeDistribution })),
+  );
+  const allEvents = allHarvests
+    .concat(allDistributions)
+    .filter((e) => BigNumber.from(e.amount).gt(ethers.constants.Zero))
+    .sort((a, b) => b.timestamp - a.timestamp);
+
+  let harvestReport = `${vault.name} Harvest Report\n`;
+
+  let totalHarvested = 0;
+  let totalAccumulated = 0;
+  let totalVaultTokens = 0;
+  let totalVaultPrincipal = 0;
+
+  console.log(`Evaluating ${allEvents.length} events for ${vault.name}`);
+  for (const event of allEvents) {
+    const token = await getFullToken(chain, event.token);
+    const amount = formatBalance(event.amount, token.decimals);
+    const { price } = await queryPriceAtTimestamp(token.address, event.timestamp * 1000);
+
+    const tokenEarned = price * amount;
+    totalAccumulated += tokenEarned;
+    if (event.type === HarvestType.Harvest) {
+      totalHarvested += amount;
+    }
+
+    let balance = 0;
+    if (isInfluenceVault(vault.address)) {
+      balance = await getInfuelnceVaultYieldBalance(chain, vault.address, event.block);
+    } else {
+      const vaultContract = Vault__factory.connect(vault.address, sdk.provider);
+      const totalSupply = await vaultContract.totalSupply({ blockTag: event.block });
+      balance = formatBalance(totalSupply);
+    }
+    const { price: vaultPrice } = await queryPriceAtTimestamp(vault.address, event.timestamp * 1000);
+    totalVaultTokens += balance;
+    totalVaultPrincipal += vaultPrice * balance;
+
+    harvestReport = harvestReport.concat(
+      `[${new Date(event.timestamp * 1000).toDateString()}] ${amount.toFixed(2)} ${
+        token.symbol
+      } ($${tokenEarned.toFixed(2)}) - Balance: ${balance.toFixed(2)}\n`,
+    );
+  }
+
+  const duration = VAULT_TWAY_PERIOD * ONE_DAY_MS;
+  const averagePrincipal = totalVaultPrincipal / allEvents.length;
+  const averageTokens = totalVaultTokens / allEvents.length;
+  const apr = calculateYield(averagePrincipal, totalAccumulated, duration);
```

Comment:
we calculate (for reporting purposes) the evaluated apr using the yield functions